Rework Code Base for Single Instance

JVickery-TBS opened the pull request.

JVickery-TBS committed:
removal(dev): tear it out;

- Tearing out all unnecessary code.

JVickery-TBS committed:
removal(dev): tear it out;

- Tearing out all unnecessary code.

JVickery-TBS committed:
Merge branch 'refactor/is-registry' into feature/single-instance

# Conflicts:
#	ckanext/canada/tests/test_trim_package.py
### RESOLVED.

JVickery-TBS committed:
Merge branch 'refactor/is-registry' into feature/single-instance

# Conflicts:
#	ckanext/canada/tests/test_trim_package.py
### RESOLVED.

JVickery-TBS committed:
Merge branch 'refactor/is-registry' into feature/single-instance

JVickery-TBS committed:
feat(dev): validators, facets, syntax;

- Syntax fixes.
- Typing fixes.
- Started working on the dataset visibility validator.
- Added more validation to portal release date field.
- Changed the published search facet to capacity.
- Set capacity=private for non standard dataset types.

JVickery-TBS committed:
feat(dev): validators;

- After validator for dataset visibility.
- Added more sysadmin UI for user publishing perms.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
fix(logic): user show;

- Side effect free.

JVickery-TBS committed:
removal(js): webassets;

- Removed geozoom.

JVickery-TBS committed:
fix(dev): misc logic;

- Fix different logic for single instance.

JVickery-TBS committed:
fix(dev): misc logic;

- Fix different logic for single instance.

JVickery-TBS committed:
feat(dev): everything;

- Force True for non portal package types.
- Override activity stuff for logged out users to show a dummy user.
- Minor template UI fixes.
- Added UI for private/public packages.
- Removed old logic methods.
- Removed anon actions.
- Use fq_list instead of complicated fq for package search.
- Handled remaining search issues between registry and portal.
- Added helper cli subcommand to set non portal package types to private.
- Added more bootstrap4+ classes.

JVickery-TBS committed:
feat(dev): some TODOs;

- Improved some code.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
fix(dev): language switcher;

- Language switcher for language domains.
- Minor remplate fixes.
- Pinned required dependency versions.

JVickery-TBS committed:
fix(logic): dataset search;

- Try/catch for dataset search.

JVickery-TBS committed:
fix(syntax): lint;

- Flake8 fix.

wardi commented on ckanext/canada/templates/base.html:
```diff
@@ -56,10 +56,7 @@
-    {# Remove when WET v4.0.21 is out #}
-    {% asset 'canada_internal/css' %}
-  {% endif %}
+  {% asset 'canada_internal/css' %}
```
These could be merged at some point

wardi reviewed (commented):
looks good, just a few minor suggestions

JVickery-TBS committed:
refactor(plugins): limit plugin count;

- Reduce plugins.

JVickery-TBS committed:
refactor(plugins): limit plugin count;

- Reduce plugins.

JVickery-TBS commented on ckanext/canada/templates/user/list.html:
```diff
@@ -59,6 +59,11 @@ <h2 class="wb-inv">{{ _('Users') }}</h2>
+                      {% if user.default_dataset_visibility == 'public' or (user.User and user.User.plugin_extras and user.User.plugin_extras['default_dataset_visibility'] == 'public') %}
+                        <span class="text-muted" title="{{- _('Creates Public Datasets') -}}"><i class="fa fa-eye"></i>&nbsp;</span>
+                      {% else %}
+                        <span class="text-muted" title="{{- _('Creates Private Datasets') -}}"><i class="fa fa-eye-slash"></i>&nbsp;</span>
```
Yessss that makes more sense for our sysadmins

JVickery-TBS committed:
refactor(plugins): limit plugin count;

- Reduce plugins.

JVickery-TBS committed:
refactor(plugins): limit plugin count;

- Reduce plugins.

JVickery-TBS committed:
refactor(assets): improve webassets;

- Use filters.
- Combine internal and public.

JVickery-TBS committed:
refactor(assets): improve webassets;

- Use filters.
- Combine internal and public.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/cli.py
#	ckanext/canada/tests/conftest.py
#	ckanext/canada/view.py
### RESOLVED.

JVickery-TBS committed:
feat(css,js): build assets;

- Start creating build front end assets.

JVickery-TBS committed:
feat(dev): npm build scripts;

- Sass and Gulp Terser building and watching.

JVickery-TBS committed:
fix(github): gulp build;

- Install and build frontend dependencies.

JVickery-TBS committed:
refactor(css): sassify;

- Moved canada custom css to scss;
- Started working on custom datatables css to scss.
- Removed source mapping from gulp.

JVickery-TBS committed:
refactor(css): sassify;

- Finished sass for datatables.

JVickery-TBS committed:
refactor(js): file prefix;

- Removed registry file prefix for most JS scripts.
-

JVickery-TBS committed:
fix(tests): id;

- Do not reuse ids.

JVickery-TBS committed:
feat(js,css): track build files;

- Track build files for deployment.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/cli.py
#	requirements.txt
### RESOLVED.

JVickery-TBS committed:
fix(dev): middleware;

- Fix missing inheritance.

JVickery-TBS committed:
fix(logic): dev;

- Package search ignore auth;

JVickery-TBS committed:
fix(types): typing;

- Pyright fixes.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
feat(templates): check activity plugin;

- Check if activity plugin is loaded.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	requirements.txt
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
temp(misc): schema;

- Default value for resource type.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/assets/public/canada_public.css
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/assets/custom/source/package_confirm_contact_email.js
#	ckanext/canada/assets/internal/canada_internal.css
#	ckanext/canada/assets/internal/webassets.yml
#	ckanext/canada/plugin/form_plugin.py
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/plugin/form_plugin.py
#	ckanext/canada/plugin/internal_plugin.py
#	ckanext/canada/templates/user/read_base.html
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/tests/test_validators.py
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/migration/canada_logic/versions/16ed91c52021_.py
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/migration/canada_logic/versions/001_0ef791477ff0_create_package_sync_table.py
#	ckanext/canada/migration/canada_logic/versions/002_16ed91c52021_set_res_positions.py
#	ckanext/canada/migration/canada_logic/versions/003_015d2fdd075f_create_res_position_constraint.py
#	ckanext/canada/migration/canada_logic/versions/0ef791477ff0_.py
#	ckanext/canada/migration/canada_logic/versions/16ed91c52021_.py
#	ckanext/canada/migration/canada_public/versions/0ef791477ff0_.py
#	ckanext/canada/migration/canada_public/versions/16ed91c52021_.py
#	ckanext/canada/plugin/public_plugin.py
### RESOLVED.

JVickery-TBS committed:
feat(migrate): new alembic;

- Fix merge conflicts of migration scripts.

JVickery-TBS committed:
fix(logic): middleware;

- Fix middleware registration.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/plugin/public_plugin.py
#	ckanext/canada/plugin/security_plugin.py
### RESOLVED.

JVickery-TBS committed:
feat(dev): ref tables;

- Started working on reference tables for pd data.

JVickery-TBS committed:
Merge branch 'master' into feature/ds-reference-tables

JVickery-TBS committed:
Merge branch 'master' into feature/ds-reference-tables

JVickery-TBS committed:
feat(dev): ds ref tables cont;

- Continued script to generate service reference data.
- Continued schema for ref tables.
- Improved pd subcommand for loading ref data.

JVickery-TBS committed:
feat(dev): ds ref tables cont;

- Finalized script to generate service ref data.
- Added more recombinant schema key/values.

JVickery-TBS committed:
feat(dev): ds ref tables cont;

- Made JSONB index.
- Added choices_filter_query.
- Added recombinant org_name to the temp ds table.
- Started writing the database validation.

JVickery-TBS committed:
feat(dev): ds ref tables cont;

- Finalized ref value checks and errors in the db func.
- Fixed up some other logic things.

JVickery-TBS committed:
feat(dev): ds fiscal year func;

- Started working on fiscal year function.

JVickery-TBS committed:
feat(dev): ds fiscal year func;

- Finalized fiscal year db function.

JVickery-TBS committed:
feat(misc): changelog;

- Added change log file.

JVickery-TBS committed:
fix(misc): syntax;

- Pyright fixes.
- Flake8 fixes.

JVickery-TBS committed:
feat(pd): service inventory;

- Made service name fields computed ones.
- Modified filter scripts for service name lookups.

JVickery-TBS committed:
fix(pd): service inventory;

- Filter script typo.

JVickery-TBS committed:
fix(pd): service inventory;

- Filter script typo.

JVickery-TBS committed:
feat(dev): github action;

- Created new workflow to semi-automate stuffs.

JVickery-TBS committed:
fix(tests): service inventory;

- Removed service name computed fields.

JVickery-TBS committed:
fix(pd): service inventory;

- Trigger type cast.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/assets/public/canada_public.css
#	ckanext/canada/plugin/public_plugin.py
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
fix(pd): service inventory;

- Trigger type cast.

JVickery-TBS committed:
fix(dev): resource views;

- Resource blueprints.

JVickery-TBS committed:
feat(schema): ckanapi dump;

- Ignore datastore fields from ckanapi dump.
- Ignore resource views from ckanapi dump.

wardi commented on ckanext/canada/schemas/presets.yaml:
```diff
@@ -4291,3 +4302,23 @@ presets:
+    required: false
+    validators: ignore
+    form_panel: _exclude_from_form
+    output_validators: canada_output_none
```
It would be much better for ckanapi to remove these fields before passing them to ckan, since they're not real fields and it's the tool generating them in the first place

RabiaSajjad committed:
update validation for agreement_value >=0

RabiaSajjad committed:
add changelog for PR# 1674

JVickery-TBS committed:
Merge branch 'grants-agreement-value' into feature/single-instance

wardi committed:
fixes for loading service.csv

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

wardi committed:
fixes for loading service-std.csv

wardi committed:
compatible program_id ORDER BY

JVickery-TBS committed:
feat(dev): misc improvements;

- Various improvements for single instance.

JVickery-TBS committed:
feat(dev): misc improvements;

- Various improvements for single instance.

JVickery-TBS committed:
feat(db): ref data hash;

- Add db table and logic for reference data file hashes during load.

JVickery-TBS committed:
revert(schema): ckanapi dump;

- Do not need the schema for ckanapi dump special fields anymore.

JVickery-TBS committed:
feat(misc): syntax;

- Lint fixes.

JVickery-TBS committed:
Merge branch 'feature/package-type-display' into feature/single-instance

# Conflicts:
#	ckanext/canada/assets/public/canada_public.css
#	ckanext/canada/templates/snippets/package_item.html
### RESOLVED.

JVickery-TBS committed:
Merge branch 'feature/embed-host-list' into feature/single-instance

# Conflicts:
#	ckanext/canada/assets/public/canada_public.css
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
fix(css): portal button;

- Syntax error.

JVickery-TBS committed:
Merge branch 'feature/embed-host-list' into feature/single-instance

JVickery-TBS committed:
Merge branch 'feature/sysadmin-ui-default-user-limit' into feature/single-instance

JVickery-TBS committed:
Merge branch 'feature/package-type-display' into feature/single-instance

# Conflicts:
#	ckanext/canada/assets/public/canada_public.css
### RESOLVED.

JVickery-TBS committed:
feat(templates): misc fixes;

- Fixes to Portal sections.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/helpers.py
#	ckanext/canada/plugin/config_declaration.yaml
#	ckanext/canada/plugin/theme_plugin.py
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
feat(dev): templates/views;

- Minor dev things, and dcat things.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/assets/public/canada_public.css
### RESOLVED.

JVickery-TBS committed:
Merge branch 'feature/user-opt-ins' into feature/single-instance

# Conflicts:
#	ckanext/canada/logic.py
#	ckanext/canada/plugin/public_plugin.py
#	ckanext/canada/plugin/theme_plugin.py
### RESOLVED.

JVickery-TBS committed:
feat(build): gulp;

JVickery-TBS committed:
feat(templates): facets;

- Reworking some of the facets to make sense with Registry Only package types.
- Improved some layouts.
- Added a Registry Only label.

JVickery-TBS committed:
feat(dev): search params;

- Added facet queries for the ready_to_publish facet field.
- Added facet query mapping to facet fields.

JVickery-TBS committed:
fix(templates): autocomplete;

- Fixed major issue with the ajax autocomplete not working.

JVickery-TBS committed:
fix(tests): domain languages;

- Started test coverage for the domain_languages plugin

JVickery-TBS committed:
fix(tests): domain languages;

- Started test coverage for the domain_languages plugin

JVickery-TBS committed:
feat(tests): domain languages;

- Started test coverage for the domain_languages plugin

JVickery-TBS committed:
feat(tests): domain languages cont;

- Continued test coverage for the domain_languages plugin.

JVickery-TBS committed:
feat(tests): domain languages cont;

- Continued test coverage for the domain_languages plugin.

JVickery-TBS committed:
feat(tests): domain languages cont;

- Continued test coverage for the domain_languages plugin.

JVickery-TBS committed:
feat(tests): domain languages cont;

- Continued test coverage for the domain_languages plugin.

JVickery-TBS committed:
feat(tests): domain languages cont;

- Continued test coverage for the domain_languages plugin.
- Lint fixes.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/helpers.py
#	ckanext/canada/logic.py
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/helpers.py
#	ckanext/canada/logic.py
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/helpers.py
#	ckanext/canada/logic.py
### RESOLVED.

JVickery-TBS committed:
fix(misc): syntax;

- Flake8 fixes.

JVickery-TBS committed:
feat(tests): repo config;

- Add in a dependency branch config.

JVickery-TBS committed:
feat(tests): repo config;

- Add in a dependency branch config.

JVickery-TBS committed:
debug github wokflow;

JVickery-TBS committed:
debug github wokflow;

JVickery-TBS committed:
debug github wokflow;

JVickery-TBS committed:
debug github wokflow;

JVickery-TBS committed:
debug github wokflow;

JVickery-TBS committed:
debug github wokflow;

JVickery-TBS committed:
debug github wokflow;

JVickery-TBS committed:
debug github wokflow;

JVickery-TBS committed:
debug github wokflow;

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
fix(tests): test app host;

- Have to specify HTTP_HOST for test app contexts.

JVickery-TBS committed:
debug github wokflow;

JVickery-TBS committed:
debug github wokflow;

JVickery-TBS committed:
debug github wokflow;

JVickery-TBS committed:
debug github wokflow;

JVickery-TBS committed:
debug github wokflow;

JVickery-TBS committed:
debug github wokflow;

JVickery-TBS committed:
debug github wokflow;

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
Merge branch 'master' into feature/ds-reference-tables

# Conflicts:
#	ckanext/canada/helpers.py
#	test-core.ini
### RESOLVED.

JVickery-TBS committed:
Merge pull request #1676 from open-data/ds-reference-tables-fixes

fixes for loading service.csv

JVickery-TBS committed:
Merge remote-tracking branch 'origin/feature/ds-reference-tables' into feature/ds-reference-tables

JVickery-TBS committed:
feat(tests): coverage;

- Added test coverage for service inventory fiscal year function.
- Added test coverage for service inventory org based fields.

JVickery-TBS committed:
fix(syntax): lint;

- Flake8 fixes.

JVickery-TBS committed:
fix(test): make test;

- Org validation for make service test.

JVickery-TBS committed:
fix(test): make test;

- Org validation for make service test.

JVickery-TBS committed:
fix(test): make test;

- Org validation for make service test.

JVickery-TBS committed:
fix(test): debug;

- Trying to fix `Prep Test Coverage Split File`

JVickery-TBS committed:
fix(test): debug;

- Trying to fix `Prep Test Coverage Split File`

JVickery-TBS committed:
Merge branch 'master' into feature/ds-reference-tables

# Conflicts:
#	ckanext/canada/tests/filters.py
#	ckanext/canada/tests/test_service.py
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/ds-reference-tables

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/plugin/theme_plugin.py
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
Merge branch 'master' into feature/ds-reference-tables

JVickery-TBS committed:
feat(comments,build): script fix, ref data;

- Rebuild service inventory ref data.
- Add fixme comment for overwritting dict assignment.

JVickery-TBS committed:
Merge branch 'master' into feature/ds-reference-tables

JVickery-TBS committed:
fix(bin): service ref data gen;

- Support multiple org names for one umd.

JVickery-TBS committed:
feat(db): max length new lines;

- Do not count new lines in max char func.
- Added test coverage.

JVickery-TBS committed:
feat(db,pd): recombinant markers, service suffixes;

- Added functionality for the new recombinant CLI markers.
- Added choice suffix to service inventory program_id.
- Added test coverage.
- Added support for old choice values in the recombinant webforms.
- If `service_migration` marker is set, invalid program_ids will be suffixed with -INV.

JVickery-TBS committed:
Merge branch 'master' into feature/ds-reference-tables

JVickery-TBS committed:
refactor(db): datastore temp table;

- Renamed the datastore_user temp table to be more accurate as an app_context table.

JVickery-TBS committed:
feat(js): pd datatables;

- Better handle legacy choices.
- Better handle choice suffixing.
- Handle insecure firefox preventing window.confirm

JVickery-TBS committed:
feat(misc): changelog;

- Added change log files.

JVickery-TBS committed:
fix(lint): syntax;

- Pyright fixes.
- Flake8 fixes.

JVickery-TBS committed:
fix(lint): syntax;

- Pyright fixes.
- Flake8 fixes.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/assets/internal/canada_internal.css
#	ckanext/canada/assets/public/canada_public.css
#	ckanext/canada/templates/admin/portal_sync.html
### RESOLVED.

JVickery-TBS committed:
refactor(dev): feedback;

- Various changes from feedback.
- Change contextual datastore flags to just be one text[] db field.

JVickery-TBS committed:
feat(pd): suffix choices;

- Improved suffix choices.
- Minor logic changes for service inventory suffixes.
- Fix newlines not displaying in pd datatables.

JVickery-TBS committed:
fix(tests): service filter;

- Pytests don't like f string.

JVickery-TBS committed:
fix(syntax): lint;

- Flake8 fixes.
- Test fix for db func change.
- Pyright fixes.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/assets/public/canada_public.css
#	ckanext/canada/plugin/dataset_plugin.py
#	ckanext/canada/plugin/form_plugin.py
#	ckanext/canada/schemas/doc.yaml
#	ckanext/canada/validators.py
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	.github/workflows/pytest.yml
#	ckanext/canada/assets/internal/canada_internal.css
#	ckanext/canada/assets/public/canada_public.css
#	ckanext/canada/plugin/dataset_plugin.py
#	ckanext/canada/plugin/internal_plugin.py
#	ckanext/canada/plugin/public_plugin.py
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/assets/public/canada_public.css
#	ckanext/canada/plugin/public_plugin.py
### RESOLVED.

JVickery-TBS committed:
fix(tests): domain map;

- Add correct domain to recombinant redirect tests.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
dev

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/tests/test_webforms.py
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
Merge branch 'master' into feature/ds-reference-tables

# Conflicts:
#	.github/workflows/pytest.yml
#	ckanext/canada/assets/public/canada_public.css
### RESOLVED.

JVickery-TBS committed:
Merge branch 'master' into feature/ds-reference-tables

JVickery-TBS committed:
Merge branch 'master' into feature/ds-reference-tables

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/assets/public/canada_public.css
### RESOLVED.

JVickery-TBS committed:
feat(schema): logic;

- Bypass validation during migrations.

JVickery-TBS committed:
Merge branch 'feature/ds-reference-tables' into feature/single-instance

# Conflicts:
#	ckanext/canada/assets/datatables/pd_datatable.css
#	ckanext/canada/assets/public/canada_public.css
#	ckanext/canada/helpers.py
#	ckanext/canada/logic.py
#	ckanext/canada/migration/canada_logic/versions/004_28a9be077875_create_ref_data_hash_tables.py
#	ckanext/canada/model.py
#	ckanext/canada/pd.py
#	ckanext/canada/plugin/internal_plugin.py
### RESOLVED.

JVickery-TBS committed:
fix(misc): syntax;

- Flake8 fixes.

JVickery-TBS committed:
feat(js): exit site warning;

- Added script to warn of leaving canada.ca domainspace.

JVickery-TBS committed:
feat(misc): changelog;

- Added change log file.

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

JVickery-TBS committed:
Merge branch 'feature/leaving-site-modal' into feature/single-instance

JVickery-TBS committed:
Merge branch 'master' into feature/single-instance

# Conflicts:
#	ckanext/canada/plugin/form_plugin.py
#	ckanext/canada/validators.py
### RESOLVED.
